3.preprocess-features/README.md

```diff
@@ -0,0 +1,54 @@
+# 3. Preprocess Features
+
+In this module, we present our pipeline for preprocessing features.
+
+### Feature Preprocessing
+
+We use [PyCytominer](https://github.com/cytomining/pycytominer) to compile single-cell features.
+
+We use [sklearn.preprocessing.StandardScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html) to derive a normalizion scaler from all negative control features.
+`StandardScaler()` standardizes features by removing the mean and scaling to unit variance.
+
+[Caicedo et al, 2017](https://www.nature.com/articles/nmeth.4397) explain why the negative control features are a good normalization population for our use case:
+> When choosing the normalizing population, we suggest the use of control samples (assuming that they are present in sufficient quantity), because the presence of dramatic phenotypes may confound results. This procedure is good practice regardless of the normalization being performed within plates or across the screen.
+
+The `get_negative_control_index_df` function inside [preprocess-features-utils.py](preprocess-features-utils.py) creates a DeepProfiler-style `index.csv` file (saved in [norm_pop_index.csv](norm_pop_index.csv)) with only the negative control wells.
+We then use pycytominer to compile the features from these negative control wells and derive and `StandardScaler` with only the negative control features (saved in [negative_control_scaler.save](negative_control_scaler.save)).
+
+After deriving a normalization scaler, we load features by plate and apply the normalization scaler to these plate features.
+The normalized features from each plate are saved in `output_path/`.
+It is necessary to normalize features by plate because we are unable to load the single-cell features for the entire screen into memory.
+
+**Note:** The single-cell dataframes for each plate are first saved as uncompressed files with `pandas.to_csv`.
+These uncompressed files are then compressed with [pigz-python](https://github.com/nix7drummer88/pigz-python) and the original, uncompressed files are deleted.
+`Pigz-python` is the python implementation of [pigz](https://zlib.net/pigz/), which uses multiple cores to compress files.
+This is significantly faster than the native `pandas` gzip compression.
```

Comment:
oh interesting! Let's leave it as is for now then (although I will note that an initial time save might not be better than the risk of a future deprecated python package (`pigz-python` doesn't appear to be widely used))

Reply:
How much faster is "significantly"?

Reply:
Can't remember the exact timings for each method, but after discussing with Jenna we have decided to stick with the native compression with `pandas` and not use `pigz-python`.

We agree there is a concern with using a package that might become deprecated in the future.